rust/lanzatool/src/generation.rs

```diff
@@ -1,47 +1,71 @@
+use serde::{Deserialize, Serialize};
+
 use std::fmt;
 use std::fs;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 use anyhow::{Context, Result};
+use bootspec::BootJson;
+use bootspec::SpecialisationName;
+use bootspec::generation::Generation;
+
+// TODO: actually, I'm not sure it's a good thing to have Default
+// we should maybe have TryDefault?
+// discuss this with upstream.
```

Comment:
Can you elaborate on the underlying issue here?

Reply:
First impression is that `Default` could be potentially harmful for specially crafted extensions where a default `PathBuf` or `String` might be "wrong" and unexpected.

But then again, the downstream consumer would probably want to validate the extension's inputs rather than blindly accept everything.

Reply:
Empty string `os_release` could be accepted due to this trait bound, I agree with @cole-h 
So my intuition is either:

- we remove `Default` and RFC-0125 is modified to ensure that `extensions` *always* exist (even if it's a empty HashMap, which is a valid representation)
- either, we do slight changes to ensure we can accept no extension when the type is HashMap (the idea behind TryDefault).
- another option?

Reply:
Couldn't we just make the field optional upstream?

Reply:
We're of the same mind: https://github.com/DeterminateSystems/bootspec/pull/68#discussion_r1052493957

Reply:
This is more or less implemented with the caveats mentioned on bootspec upstream, which are reflected by the necessary unwrapping in `src/install.rs`, which feels a bit weird to me because we are doing the final validation during install rather than when we read the bootspec.

I don't feel so good about erroring this late. :/

Reply:
(and doing an explicit unwrap just after serialization seems to be equivalent to do some serde work on behalf of the library.)